mobile/src/controllers/NotificationsController.ts

```diff
@@ -254,6 +253,22 @@ export class NotificationsController implements IDisposable {
         this.settings.updateNotifications({ enabled: this._notificationsEnabledByUser }, 'enabled');
     };
 
+    public goToAffirmation(): boolean {
+        const now = new Date();
+        const lastScheduledDate = this.settings.current.notifications.lastAffirmationScheduledDate;
+
+        if ((now.getTime() - lastScheduledDate) > 86400000) {
+            if (!this.scheduleTime) {
+                this.scheduleTime = this.settings.current.notifications.scheduledTime;
+            }
+
+            const todaysScheduledDate = new Date();
+            todaysScheduledDate.setHours(this.scheduleTime.hour, this.scheduleTime.minute, 0);
+            return now > todaysScheduledDate;
+        }
+        return false;
```

Comment:
@MarkDavies0 I am just thinking of some edge cases here. What if someone changes their notification date from 3PM to 10AM? In that case, this logic will cause them to skip the affirmation if they enter the app at anytime between 10AM-3PM the day after they change their notification time. Instead of this logic, I think we can just check the current time to see if it's 24 hours past yesterday at the same time as the notification schedule time. That way, it would be more reliable. Something like:

```suggestion
const yesterdayThisTime = new Date();
yesterdayThisTime.setDate(yesterdayThisTime.getDate() - 1);

const lastScheduleDate = new Date();
lastScheduleDate.setDate(lastScheduleDate.getDate() - 1);
lastScheduleDate.setHours(this.scheduleTime.hour, this.scheduleTime.minute, 0);

return yesterdayThisTime >= lastScheduleDate;
```

Reply:
Also, could you point me to the part of your code in #174 that handles this? I just want to double-check the logic in there again @MarkDavies0 thanks!

Reply:
@wynnset see lines 288 - 305 in `NotificationsController.ts`